scripts/linters/pre_commit_linter.py

```diff
@@ -418,29 +419,52 @@ def categorize_files(file_paths):
     _FILES.update(all_filepaths_dict)
 
 
-def _print_complete_summary_of_errors(all_messages):
-    """Print complete summary of errors."""
-    error_messages = all_messages
-    if error_messages != '':
+def _print_complete_summary_of_lint_messages(lint_messages):
+    """Print complete summary of lint messages."""
+    if lint_messages != '':
         python_utils.PRINT('Summary of Errors:')
         python_utils.PRINT('----------------------------------------')
-        for message in error_messages:
+        for message in lint_messages:
             python_utils.PRINT(message)
 
 
-def _get_task_output(all_messages, task, semaphore):
+def _get_task_output(lint_messages, error_messages, task, semaphore):
     """Returns output of running tasks.
 
     Args:
-        all_messages: list(str). List of summary messages of linter output.
+        lint_messages: list(str). List of summary messages of linter output.
+        error_messages: list(str). List of error messages during linter
+            execution.
         task: object(TestingTaskSpec). The task object to get output of linter.
         semaphore: threading.Semaphore. The object that controls how many tasks
             can run at any time.
     """
-    all_messages += task.output
+    try:
+        lint_messages += task.output
+    except Exception:
+        error_messages.append(
+            python_utils.convert_to_bytes(task.stacktrace))
     semaphore.release()
```

Comment:
Why do we need the sempahore.release over here? I see TaskThread.run method already does this, right?

Reply:
Because we used semaphore one more time in Line  568 and 572, and we need to release it. And we used semaphore here because without using it the message of one file sometimes showed under another file.

Reply:
I have two questions:
1. Why we are using semaphores here? (I'm assuming this code is not running in multi-thread, right?)
2. Why we are acquiring semaphore in one method and releasing it in another method? (This looks like a random `release`)

Reply:
1. We used semaphore because we needed a locking mechanism, sot that lint errors of one file do not get printed under another file.
2. We are acquiring semaphore in one method and releasing it in another so that the function doesn’t even get put on the memory heap.([see comment](https://github.com/oppia/oppia/pull/9195#discussion_r418908092))

Reply:
Actually, I'm not too sure about this. What would you advise?

Reply:
I think we should relook into it, can you please create an issue and talk to Kevin on this (I think he was the one who has introduced this, right?)? [Make sure to assign the issue to yourself and it would be great if we can resolve it soon!]

Reply:
I think we should relook into it, can you please create an issue and talk to Kevin on this (I think he was the one who has introduced this, right?)? [Make sure to assign the issue to yourself and it would be great if we can resolve it soon!]

Reply:
This isn't suggested by the Kevin, it was in the GSoC follow up doc
![Screenshot from 2020-05-18 16-18-52](https://user-images.githubusercontent.com/29057268/82205045-8c00f700-9923-11ea-8fee-94cd5527f679.png)